config/v1/0000_10_config-operator_01_infrastructure.crd.yaml

```diff
@@ -99,6 +99,24 @@ spec:
                                 empty.
                               type: string
                               pattern: ^https://
+                      userTags:
+                        description: userTags is a list of additional tags to apply
+                          to AWS resources created for the cluster. Changes made to
+                          userTags will not be reflected in existing AWS resources.
+                          The userTags are only applied to an AWS resource when the
+                          resource is created.
+                        type: array
+                        items:
+                          description: AWSUserTag is a tag to apply to AWS resources
+                            created for the cluster.
+                          type: object
+                          properties:
+                            key:
```

Comment:
What if users want to have different tags for the machine-api and other components?  This API does not allow for this facility.

Also, having this be platform specific means we need special logic for each platform in the future.  Why would we define centralized cluster tags for one platform and not another?

Reply:
> What if users want to have different tags for the machine-api and other components? This API does not allow for this facility.

The user still retains the capability of adding additional tags to specific machinesets/machines just like they do now. Those tags would be additive with the global tags, where the global tags have precedence.

Reply:
> Those tags would be additive with the global tags, where the global tags have precedence.

This is not stated in the CRD description and should be.  Also, this runs contrary to how most config is typically applied (specific should override the general).

Also, since this mechanism is being tied IAM permissions in a particular use case, we need to make sure this mechanism is granular.